### PR TITLE
Simplification of groups api

### PR DESCRIPTION
**Pull request by jespino**

### CODE REVIEW

File: api4/group.go [hunk 1]
```diff
@@ -134,300 +129,271 @@ func patchGroup(c *Context, w http.ResponseWriter, r *http.Request) {
...
+		return
+	}
+	syncableID := c.Params.SyncableId
 
-		body, err := ioutil.ReadAll(r.Body)
-		if err != nil {
-			c.Err = model.NewAppError("Api4.createGroupSyncable", "api.group.io_error", nil, err.Error(),
-				http.StatusNotImplemented)
-		}
+	c.RequireSyncableId()
```
Comment (mkraft) [inline api4/group.go]: This should be `RequireSyncableType()`.

File: api4/group.go [hunk 2]
```diff
@@ -134,300 +129,271 @@ func patchGroup(c *Context, w http.ResponseWriter, r *http.Request) {
...
+	c.RequireGroupId()
+	if c.Err != nil {
+		return
+	}
 
-		if !c.App.SessionHasPermissionTo(c.Session, model.PERMISSION_MANAGE_SYSTEM) {
-			c.SetPermissionError(model.PERMISSION_MANAGE_SYSTEM)
-			return
-		}
+	c.RequireSyncableId()
```
Comment (mkraft) [inline api4/group.go]: And here.

File: api4/group.go [hunk 3]
```diff
@@ -134,300 +129,271 @@ func patchGroup(c *Context, w http.ResponseWriter, r *http.Request) {
...
+	if c.Err != nil {
+		return
+	}
+	syncableID := c.Params.SyncableId
 
-		if !c.App.SessionHasPermissionTo(c.Session, model.PERMISSION_MANAGE_SYSTEM) {
-			c.SetPermissionError(model.PERMISSION_MANAGE_SYSTEM)
-			return
-		}
+	c.RequireSyncableId()
```
Comment (mkraft) [inline api4/group.go]: And here.

File: api4/group.go [hunk 4]
```diff
@@ -134,300 +129,271 @@ func patchGroup(c *Context, w http.ResponseWriter, r *http.Request) {
...
+		return
+	}
+	syncableID := c.Params.SyncableId
 
-		body, err := ioutil.ReadAll(r.Body)
-		if err != nil {
-			c.Err = model.NewAppError("Api4.patchGroupSyncable", "api.group.io_error", nil, err.Error(),
-				http.StatusNotImplemented)
-		}
+	c.RequireSyncableId()
```
Comment (mkraft) [inline api4/group.go]: And here.

File: api4/group.go [hunk 5]
```diff
@@ -134,300 +129,271 @@ func patchGroup(c *Context, w http.ResponseWriter, r *http.Request) {
...
+	if c.Err != nil {
+		return
+	}
+	syncableID := c.Params.SyncableId
 
-		if !c.App.SessionHasPermissionTo(c.Session, model.PERMISSION_MANAGE_SYSTEM) {
-			c.SetPermissionError(model.PERMISSION_MANAGE_SYSTEM)
-			return
-		}
+	c.RequireSyncableId()
```
Comment (mkraft) [inline api4/group.go]: And here.

**Comment by mkraft** [comment]
Also there are some conflicts now since I added some feature flag checks.

File: api4/group.go (hunk 3 above)
Comment (jespino) [reply api4/group.go]: Yes, you are completely right.

**Pull request merged**